File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/season/model/TvSeasonAggregateCrewJob.java
package dev.reuss.tmdb.domain.tv.season.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Aggregate crew job for a TV season.
 *
 * @param creditId     credit id
 * @param job          job
 * @param episodeCount episode count
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeasonAggregateCrewJob(
        @JsonProperty("credit_id")
        String creditId,

        String job,

        @JsonProperty("episode_count")
        int episodeCount
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/season/model/TvSeasonAggregateCrewJob.kt
package dev.reuss.tmdb.domain.tv.season.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Aggregate crew job for a TV season.
 *
 * @property creditId credit id
 * @property job job
 * @property episodeCount Number of associated episodes.
 */
@JvmRecord
data class TvSeasonAggregateCrewJob(
    @all:JsonProperty("credit_id")
    val creditId: String?,
    @all:JsonProperty("job")
    val job: String?,
    @all:JsonProperty("episode_count")
    val episodeCount: Int,
) : TmdbModel
